src/dmd/mtype.h

```diff
@@ -126,6 +126,14 @@ extern unsigned char impcnvType2[TMAX][TMAX];
 // If !=0, give warning on implicit conversion
 extern unsigned char impcnvWarn[TMAX][TMAX];
 
+enum VarArg
+{
+    none     = 0,  /// fixed number of arguments
+    variadic = 1,  /// T t, ...)  can be C-style (core.stdc.stdarg) or D-style (core.vararg)
+    typesafe = 2,  /// T t ...) typesafe https://dlang.org/spec/function.html#typesafe_variadic_functions
```

Comment:
This wasn't particularly wise.  There is no scoping in C++ enums (better name is e.g: `VAtypesafe`)

Reply:
Can we start to use `enum class` now when DMC is ported to D?

Reply:
Don't think so, we still want D to work everything from Solaris 10, OSX 10.6, and FreeBSD 8.  And the bootstrap builds for gcc are explicitly `-std=gnu++98`.

Reply:
Only OSX >= 10.9 and FreeBSD == 11 are officially supported.
See e.g. https://dlang.org/changelog/2.079.0.html#osx-10_9

Reply:
That's fine for dmd, don't impede other compilers based on this however.

Reply:
Is it macOS 10.6 because that's the oldest that GCC supports?

Reply:
I think 10.3 is the oldest supported, I know they have servers/desktops running 10.4 that's occasionally checked, 10.5 is the default minimum version if you only configure for `i386-pc-darwin` (without say what version of darwin), and 10.6 is the version running on the mac machine I have.

Reply:
@ibuclaw feel free to adjust it for your needs. I don't have any C++ code that needs these .h files anymore. You could also call the members `VarArg_typesafe` on the C++ side.